src/raw/mod.rs

```diff
@@ -76,26 +216,42 @@ impl<K, V> Table<K, V> {
 
 impl<K, V> Drop for Table<K, V> {
     fn drop(&mut self) {
-        // we need to drop any forwarding nodes (since they are heap allocated).
-
         // safety: we have &mut self _and_ all references we have returned are bound to the
         // lifetime of their borrow of self, so there cannot be any outstanding references to
         // anything in the map.
         let guard = unsafe { crossbeam_epoch::unprotected() };
 
+        // since BinEntry::Nodes are either dropped by drop_bins or transferred to a new table,
+        // all bins are empty or contain a Shared pointing to shared the BinEntry::Moved (if
+        // self.bins was not replaced by drop_bins anyway)
         for bin in &self.bins[..] {
             let bin = bin.swap(Shared::null(), Ordering::SeqCst, guard);
+            // TODO: do we need the safety check against moved or do we assume the above invariant here?
```

Comment:
What safety check are you thinking of?

Reply:
That's what I mentioned in the original PR post. This loop does not actually drop anything anymore. Before, it would drop the `Moved` entries in the bin vector, but now we don't have these. So all this does currently is check that bins are either `null` or a reference to the shared `Moved`. However, this should be the case anyways as outline by the comment above the loop.